### PR TITLE
prepare-layout : gen-lib-cabal source and cabal file

### PR DESCRIPTION
**Pull request by shayne-fletcher-da**
This PR provides a program to generate a cabal file given a GHC source distribution for `ghc-lib`.

### CODE REVIEW

File: ghc-lib-gen/ghc-lib-gen.cabal [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+version:             0.1.0.0
+synopsis:            Cabal file generator for GHC as a library
+description:         Generate a cabal file for a subset of GHC for use as a library.
+homepage:            https://github.com/digital-asset/ghc-lib
+license:             BSD3
+x-license:           BSD-3-Clause OR Apache-2.0
+license-file:        LICENSE
+author:              Shayne Fletcher (shayne.fletcher@digitalasset.com)
+maintainer:          Shayne Fletcher (shayne.fletcher@digitalasset.com)
+copyright:           Digital Asset 2019
```
Comment (neil-da) [inline ghc-lib-gen/ghc-lib-gen.cabal]: 2018-2019

File: ghc-lib-gen/ghc-lib-gen.cabal [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+category:            Development
+build-type:          Simple
+extra-source-files:  ChangeLog.md
+cabal-version:       >=1.10
+
+executable ghc-lib-gen
+  main-is:             Main.hs
+  other-modules:
+  other-extensions:
+  build-depends:       base >=4.11 && <4.12, process >=1.6 && <1.7, filepath >=1.4 && <1.5, extra >=1.6 && <1.7
```
Comment (neil-da) [inline ghc-lib-gen/ghc-lib-gen.cabal]: I tend to put these on different lines, and avoid upper bounds (they rarely help)

File: ghc-lib-gen/ghc-lib-gen.cabal [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+maintainer:          Shayne Fletcher (shayne.fletcher@digitalasset.com)
+copyright:           Digital Asset 2019
+category:            Development
+build-type:          Simple
+extra-source-files:  ChangeLog.md
+cabal-version:       >=1.10
+
+executable ghc-lib-gen
+  main-is:             Main.hs
+  other-modules:
```
Comment (neil-da) [inline ghc-lib-gen/ghc-lib-gen.cabal]: You can omit this line

File: ghc-lib-gen/ghc-lib-gen.cabal [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  other-extensions:
```
Comment (neil-da) [inline ghc-lib-gen/ghc-lib-gen.cabal]: other-extensions is pretty useless

File: ghc-lib-gen/ChangeLog.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# Revision history for ghc-lib-gen
+
+## 0.1.0.0 -- YYYY-mm-dd
+
+* First version. Released on an unsuspecting world.
```
[the remark this answers is not in the document]
Comment (shayne-fletcher-da) [reply ghc-lib-gen/ChangeLog.md]: `cabal init` generates that :)

File: ghc-lib-gen/LICENSE [hunk 6]
```diff
@@ -0,0 +1,217 @@
+Copyright Digital Asset 2019.
```
[the remark this answers is not in the document]
Comment (shayne-fletcher-da) [reply ghc-lib-gen/LICENSE]: will fix.

File: ghc-lib-gen/ChangeLog.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (shayne-fletcher-da) [reply ghc-lib-gen/ChangeLog.md]: will fix.

File: README.md [hunk 7]
```diff
@@ -1 +1,3 @@
-# ghc-lib
\ No newline at end of file
+# ghc-lib
+
+`ghc-lib` enables use of the GHC compiler as a library. This Digital Asset (https://www.digitalasset.com) project is a work in progress and not quite ready for public consumption. Check back soon!
```
[the remark this answers is not in the document]
Comment (shayne-fletcher-da) [reply README.md]: ok, will rework it.

File: ghc-lib-gen/ghc-lib-gen.cabal (hunk 3 above)
Comment (shayne-fletcher-da) [reply ghc-lib-gen/ghc-lib-gen.cabal]: yes, deleted that and the other empty one.

**Pull request merged**